kratos/solving_strategies/schemes/residual_based_bossak_displacement_scheme.hpp

```diff
@@ -505,14 +512,25 @@ class ResidualBasedBossakDisplacementScheme
      * @param PreviousAcceleration The previous acceleration
      */
     inline void UpdateVelocity(
-        array_1d<double, 3>& CurrentVelocity,
-        const array_1d<double, 3>& DeltaDisplacement,
-        const array_1d<double, 3>& PreviousVelocity,
-        const array_1d<double, 3>& PreviousAcceleration
+        NodeIterator itNode,
+        array_1d<double, 3>& rCurrentVelocity,
+        const array_1d<double, 3>& rDeltaDisplacement,
+        const array_1d<double, 3>& rPreviousVelocity,
+        const array_1d<double, 3>& rPreviousAcceleration
         )
     {
-        noalias(CurrentVelocity) = (mBossak.c1 * DeltaDisplacement - mBossak.c4 * PreviousVelocity
-                                     - mBossak.c5 * PreviousAcceleration);
+        const array_1d<double, 3> auxiliar_updated_velocity = (mBossak.c1 * rDeltaDisplacement - mBossak.c4 * rPreviousVelocity - mBossak.c5 * rPreviousAcceleration);
+
+        // We check if the dofs are fixed
+        if (!itNode->IsFixed(VELOCITY_X)) {
```

Comment:
I don't really agree with this change:

if you correctly performed the solution, than the relation between displacement, velocity and acceleration should be guaranteed.

What i am trying to tell here is that if the velocity of the node is different from auxilar_updated_velocity it is a symptom that you are incorrectly calculating the rDeltaDisplacement

Reply:
After the last changes the predict correctly works and can be replaced, but I left for consistency reasons

Reply:
do not... this would hide errors

Reply:
It could, but then how can you solve the problem of #4426 ?

Reply:
If everything is calcualted consistently in the predict this will solve #4426 
the fact that this is not working is a symptom that there is still an inconsistency in the formulation, meaning that the displacement update is not consistent with the prescribed velocity